File: COMP9041/lab07/log_probability.py
#!/usr/bin/python

import sys
import glob
import re
import math

result = []
key = (sys.argv[1]).lower()
for poet in glob.glob("poets/*.txt"):
	count = []
	subresult = []
	F = open(poet, 'r')
	for line in F:
		line = line.lower()
		count += re.split(r"[^A-Za-z]", line)

	filename = poet.replace(".txt", "")
	filename = filename.replace("_", " ")
	filename = filename.replace("poets/", "")

	count = filter(None, count)
	length = len(count)
	keynumber = count.count(key)
	subresult.append(filename)
	subresult.append(length)
	subresult.append(keynumber)
	result.append(subresult)

result.sort()
for sub in result:
	print ("log(({:d}+1)/{:6d}) = {:8.4f} {}".format(sub[2], sub[1], math.log((float(sub[2])+1)/float(sub[1])), sub[0]))

File: COMP9041/lab07/log_probability.pl
#!/usr/bin/perl -w

$key = lc $ARGV[0];

foreach $file (glob "poets/*.txt") {
	$total = 0;
	$key_total = 0;
	open (F, "<$file") or die "Cannot open file $file\n";
	while (<F>) {
        	@words = split /[^A-Za-z]/, $_;
		foreach(@words) {
                	if (lc $_ eq $key) {
				$key_total ++;
                	}
                	if (!($_ =~ /^$/)) {
                        	$total ++;
			}
        	}
	}

	$file =~ s/poets\///;
	$file =~ s/\.txt//;
	$file =~ tr/_/ /;
	$table{$file} = $total;
	$table{$file}{$key} = $key_total;
}

#$size = keys %table;
foreach $poet (sort keys %table) {
	printf "log((%d+1)/%6d) = %8.4f %s\n", $table{$poet}{$key}, $table{$poet}, log(($table{$poet}{$key} + 1)/$table{$poet}), $poet;
}

